mne/io/pick.py

```diff
@@ -188,6 +188,10 @@ def pick_types(info, meg=True, eeg=False, stim=False, eog=False, ecg=False,
     """
     # NOTE: Changes to this function's signature should also be changed in
     # PickChannelsMixin
+    from .meas_info import Info
+    if not isinstance(info, Info):
+        raise TypeError('info must be an instance of Info, not %s'
+                        % type(info))
```

Comment:
FYI I made this change because we had no check to ensure `info` was an instance of `Info` here before, and I got some weird error when I accidentally passed `raw` instead of `raw.info`. We could relax this check to `isinstance(info, (Info, dict))` but I'd prefer to keep it more strict to `Info` for now to help ensure people create their info structs correctly by reading or by using `create_info`. This necessitated a couple of minor changes elsewhere to comply with this check.